<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Camera parameters -->
  <arg name="rgb_image_topic" default="kinect2/qhd/image_color_rect"/>
  <arg name="rgb_camera_info_topic" default="kinect2/qhd/camera_info"/>
  <arg name="depth_image_topic" default="kinect2/qhd/image_depth_rect"/>
  <arg name="depth_camera_info_topic" default="kinect2/qhd/camera_info"/>

  <arg name="global_frame_id" default="map"/>
  <arg name="base_frame_id" default="base_footprint"/>

  <!-- Detector model parameters -->
  <arg name="detector_model_filename" default="$(find uwds3)/models/detection/ssd_mobilenet_v2_coco_2018_03_29.pbtxt"/>
  <arg name="detector_weights_filename" default="$(find uwds3)/models/detection/ssd_mobilenet_v2_coco_2018_03_29.pb"/>
  <arg name="detector_config_filename" default="$(find uwds3)/config/detection/mscoco_config.yaml"/>

  <!-- Tracker parameters -->
  <arg name="n_init" default="1"/>
  <arg name="n_frame" default="3"/>
  <arg name="max_iou_distance" default="0.98"/>
  <arg name="max_color_distance" default="0.3"/>
  <arg name="max_face_distance" default="0.3"/>
  <arg name="max_centroid_distance" default="0.4"/>
  <arg name="max_lost" default="5"/>
  <arg name="max_age" default="15"/>

  <arg name="use_word_embeddings" default="False"/>
  <arg name="static_word_embeddings_filename" default="$(find uwds3)/models/word_embeddings/wiki-news-300d-1M-subword.vec"/>

  <!-- Simulation parameters -->
  <arg name="use_simulation_gui" default="False"/>
  <arg name="simulation_config_filename" default="$(find uwds3)/config/simulation/pr2_config.yaml"/>
  <arg name="cad_models_additional_search_path" default="$(find uwds3)/models/cad_models"/>
  <arg name="static_entities_config_filename" default="$(find uwds3)/config/simulation/empty_floor.yaml"/>
  <arg name="robot_urdf_file_path" default="urdf/pr2.urdf"/>

  <!-- Other parameters -->
  <arg name="publish_viz" default="true"/>
  <arg name="publish_tf" default="true"/>
  <arg name="use_depth" default="false"/>

  <node name="navigation_pipeline" pkg="uwds3" type="navigation_pipeline_node.py" args="" output="screen" respawn="false">
    <rosparam subst_value="true">

      rgb_image_topic: $(arg rgb_image_topic)
      rgb_camera_info_topic: $(arg rgb_camera_info_topic)

      depth_image_topic: $(arg depth_image_topic)
      depth_camera_info_topic: $(arg depth_camera_info_topic)

      global_frame_id: $(arg global_frame_id)
      base_frame_id: $(arg base_frame_id)

      detector_model_filename: $(arg detector_model_filename)
      detector_weights_filename: $(arg detector_weights_filename)
      detector_config_filename: $(arg detector_config_filename)

      use_word_embeddings: $(arg use_word_embeddings)
      static_word_embeddings_filename: $(arg static_word_embeddings_filename)

      use_simulation_gui: $(arg use_simulation_gui)
      simulation_config_filename: $(arg simulation_config_filename)
      cad_models_additional_search_path: $(arg cad_models_additional_search_path)
      static_entities_config_filename: $(arg static_entities_config_filename)
      robot_urdf_file_path: $(arg robot_urdf_file_path)

      publish_viz: $(arg publish_viz)
      publish_tf: $(arg publish_tf)

      use_depth: $(arg use_depth)

      n_init: $(arg n_init)
      max_iou_distance: $(arg max_iou_distance)
      max_color_distance: $(arg max_color_distance)
      max_face_distance: $(arg max_face_distance)
      max_lost: $(arg max_lost)
      max_age: $(arg max_age)
    </rosparam>
  </node>

</launch>
